internal/testutil/minio.go

```diff
@@ -1,103 +1,64 @@
 package testutil
 
 import (
-	"bufio"
-	"context"
 	"fmt"
-	"io"
 	"testing"
 
 	"github.com/minio/minio-go/v7"
 	"github.com/minio/minio-go/v7/pkg/credentials"
-	"github.com/ory/dockertest/v3"
-	"github.com/ory/dockertest/v3/docker"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
 )
 
 // WithTestMinIO starts a test MinIO server
-func WithTestMinIO(t *testing.T, bucket string, handler func(endpoint string) error) error {
+func WithTestMinIO(t *testing.T, bucket string, handler func(endpoint string)) {
 	t.Helper()
 
-	ctx, clearTimeout := context.WithTimeout(context.Background(), maxWait)
-	defer clearTimeout()
-
-	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
-	pool, err := dockertest.NewPool("")
+	ctx := GetContext(t, maxWait)
+
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Name:         "pomerium-minio",
+			Image:        "quay.io/minio/minio:RELEASE.2022-12-02T19-19-22Z",
```

Comment:
Question: did you evaluate using the Minio module? https://golang.testcontainers.org/modules/minio/